app/reflexes/products_reflex.rb

```diff
@@ -46,6 +46,36 @@ def bulk_update
     render_products_form_with_flash
   end
 
+  def delete_product
+    id = current_id_from_element(element)
+    authorize! :delete, Spree::Product
+    product = product_finder(id).find_product
+    authorize! :delete, product
+
+    if product.destroy
+      flash[:success] = I18n.t('admin.products_v3.delete_product.success')
+    else
+      flash[:error] = I18n.t('admin.products_v3.delete_product.error')
+    end
+
+    fetch_and_render_products_with_flash
```

Comment:
I just had another thought. We are re-rendering the whole table. That means that existing edits of other products are lost without warning here. I think that we need to re-render only the one product or variant. Or we need to force people to save before they proceed.

A completely different approach would be to just mark the variant as deleted and perform the deletion when you save everything in bulk. That would be more in line with the rest of the page.

I personally prefer small instant updates but the rest of the page doesn't work that way.

Oh, alternatively, as quick fix, we can tell the user in the modal that all unsaved changes will be lost. That may be good enough for now.

Reply:
> I just had another thought. We are re-rendering the whole table. That means that existing edits of other products are lost without warning here. I think that we need to re-render only the one product or variant. Or we need to force people to save before they proceed.

Ah, that's a good catch.

> as quick fix, we can tell the user in the modal that all unsaved changes will be lost. That may be good enough for now.

Yes, I agree. That's the quick fix we should opt for now.
Hey @mariocarabotta - we need to handle this edge case. Can we get some quick design on this solution? Thanks.

Reply:
We opted to accept this case in the short term, to avoid creating further dialog messages. And have created a new issue to resolve this by another method: https://github.com/openfoodfoundation/openfoodnetwork/issues/11987